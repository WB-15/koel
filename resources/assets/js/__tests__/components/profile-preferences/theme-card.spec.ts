import { shallow } from '@/__tests__/adapter'
import Component from '@/components/profile-preferences/theme-card.vue'

describe('profile-preferences/profile-form', () => {
  afterEach(() => {
    jest.resetModules()
    jest.clearAllMocks()
  })

  const theme: Theme = {
    id: 'sample',
    thumbnailColor: '#f00'
  }

  it('renders', () => {
    expect(shallow(Component, {
      propsData: {
        theme
      }
    })).toMatchSnapshot()
  })

  it('emits an event when theme is selected', () => {
    const wrapper = shallow(Component, {
      propsData: {
        theme
      }
    })

    wrapper.click('[data-testid=theme-card-sample]')
    expect(wrapper.hasEmitted('selected', theme)).toBe(true)
  })
})
