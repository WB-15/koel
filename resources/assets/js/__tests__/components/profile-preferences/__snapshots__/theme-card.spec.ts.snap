// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile-preferences/profile-form renders 1`] = `
<div data-testid="theme-card-sample" class="theme" style="background-color: rgb(255, 0, 0);">
  <div class="name">Sample</div>
</div>
`;
